### PR TITLE
Unified tier type/order/name dropdowns into a single cascading dropdown

### PR DESCRIPTION
**Pull request by mattchrlw**
This PR also requires the corresponding PR in `elpis` to be merged.

This reconfigures the dropdowns for Import Settings/Tiers so that instead of having three dropdowns (of which only one is meant to be selected), there is a single dropdown for "Selection Mechanism" (Tier Type/Tier Name/Tier Order) which then cascades into a second dropdown based on the first.

To facilitate this a `shown` parameter was added to the setting-related methods in `elpis/transformer/__init__.py` which specifies whether a given row of the settings should be shown by default. Currently Tier Type is set to `shown=True` and Tier Name and Tier Order are set to `shown=False`.

image

### CODE REVIEW

File: src/components/Dataset/GeneratedUI.js [hunk 1]
```diff
@@ -62,84 +62,83 @@ const GeneratedUI = ({settings, ui, changeSettingsCallback}) => {
...
+                            }
+                            break;
+                        case 'select': {
+                            let options = [];
+                            // Build options
+                            data.options.forEach(v => {
                                 options.push({key: v, value: v, text: v})
+                            });
+                            dataEntryElement = (<Select
+                                // value={settings[ui_name]}
```
Comment (benfoley) [inline src/components/Dataset/GeneratedUI.js:99]: @mattchrlw value preselects based on the dataset setting value. Might just need to uncomment this. I'll try it later

File: src/components/Dataset/GeneratedUI.js [hunk 2]
```diff
@@ -62,84 +62,90 @@ const GeneratedUI = ({settings, ui, changeSettingsCallback}) => {
...
+                                        }
+                                        ui['data'][data.value]['shown'] = true;
+                                        // Update new settings with default value of selected mechanism
+                                        newSettings[data.value] = ui['data'][data.value]['options'][0]
+                                    }
+                                    changeSettingsCallback(newSettings)
+                                }}
+                                selection
+                            />);
+                            // TODO: add a onChange that dispatches the setting (do this for int and string as well)
```
Comment (nicklambourne) [inline src/components/Dataset/GeneratedUI.js:133]: Does this need to be addressed for this PR?

**Review by nicklambourne — CHANGES_REQUESTED** [review]
Issue:
- Set up with first 1/4 of man corpus
- Select tier_order, switch back to tier_name
- Select tx as tier name. UI does not update. I suspect state updates, but selection in dropdown does not. Can be corrected by selecting tx again, but this should be fixed now.

File: src/components/Dataset/GeneratedUI.js [hunk 3]
```diff
@@ -62,84 +62,90 @@ const GeneratedUI = ({settings, ui, changeSettingsCallback}) => {
...
+                            // Build options
+                            data.options.forEach(v => {
+                                options.push({key: v, value: v, text: v})
+                            });
+                            if (ui_name === "selection_mechanism") {
+                                settings[options[0].value] = ui['data'][options[0].value]['options'][0];
+                            }
+                            dataEntryElement = (<Select
                                 value={settings[ui_name]}
+                                // value={ui_name === "selection_mechanism"
```
[the remark this answers is not in the document]
Comment (mattchrlw) [reply src/components/Dataset/GeneratedUI.js:103]: That was for testing, removed

File: src/components/Dataset/GeneratedUI.js (hunk 2 above)
Comment (mattchrlw) [reply src/components/Dataset/GeneratedUI.js:134]: probably not needed for now

**Review by nicklambourne — CHANGES_REQUESTED** [review]
Other selection mechanisms no longer appear and neither do the punctuation and cleaning option fields:
Screen Shot 2020-08-27 at 1 11 51 am

**Review by nicklambourne — APPROVED** [review]

**Pull request merged**